### PR TITLE
[libs/client] Add chainwebjs to kadenajs

### PR DESCRIPTION
**Pull request by nillo**
## Reason: Added type support for chainwebJS, reason types where missing. enables chainwebJS to be part of kadenaJS until new api has been created, and a replacement package for chainweb has to be written. 

* This package is a temporary solution and will be deprecated once the new version for the new api comes out.
* Minimal code tweaks have been made to original, mostly types added in agreement with randy

## Check off the following:

- [x] I have reviewed my changes and run the appropriate tests.
- [x] I have have run `rush change` to add the appropriate change logs.
- [x] I have added/edited docs.
- [x] I have added tutorials.
- [x] I have double checked and DEFINITELY added docs.

### CODE REVIEW

File: packages/libs/chainwebjs/src/paging.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/* ************************************************************************** */
+/* Pageing Tools */
```
Comment (ash-vd) [inline packages/libs/chainwebjs/src/paging.ts:2]:
```suggestion
/* Paging Tools */
```

**Review by ash-vd — APPROVED** [review]

File: packages/libs/chainwebjs/src/tests/mocks/360hashes.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable max-lines */
```
Comment (alber70g) [inline packages/libs/chainwebjs/src/tests/mocks/360hashes.ts:1]: Let's add the `tests/mocks` directory to `packages/libs/chainwebjs/.eslintignore`

File: packages/libs/chainwebjs/src/tests/retries.test.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import chainweb from '..';
+import { parseResponse } from '../internal';
+import { chainUrl } from '../request';
+
+/* ************************************************************************** */
+/* Retries */
+
+describe('retry', () => {
+  test('retry 404', async () => {
```
Comment (alber70g) [inline packages/libs/chainwebjs/src/tests/retries.test.ts:9]: Can we change all `test` to `it`?
Convention is to use `describe-it` syntax.

File: packages/libs/chainwebjs/src/tests/transactions.test.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (alber70g) [inline packages/libs/chainwebjs/src/tests/transactions.test.ts:1]: Idem

File: packages/libs/chainwebjs/src/tests/transactions.test.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    const r = await chainweb.transaction.height(0, height);
+    logg('Transactions:', r);
+    expect(r.length).toBe(1);
+    expect(r[0]).toHaveProperty('transaction');
+    expect(r[0]).toHaveProperty('output');
+    expect(r[0]).toHaveProperty('height');
+    expect(r[0].height).toBe(height);
+  });
+});
+
```
Comment (alber70g) [inline packages/libs/chainwebjs/src/tests/transactions.test.ts:81]: The describes in this file should probably be wrapped in a

`describe('chainweb.transactions`, () => {`

with `it`s being what is now in the current `describe`s

File: packages/libs/chainwebjs/src/transactions.ts [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+ */
+export async function recentTxs(
+  chainId: number | string,
+  depth: number = 0,
+  n: number = 1,
+  network?: string,
+  host?: string,
+): Promise<ITransactionElement[]> {
+  const x = await recentBlocks(chainId, depth, n, network, host);
+  return filterTxs(x);
```
Comment (alber70g) [inline packages/libs/chainwebjs/src/transactions.ts:77]: Let's give the `x` a meaningful name

File: packages/libs/chainwebjs/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "./node_modules/@kadena-dev/heft-rig/profiles/default/tsconfig-base.json",
+  "compilerOptions": {
+    "types": ["heft-jest", "node"],
```
Comment (alber70g) [inline packages/libs/chainwebjs/tsconfig.json:4]: Can we get rid of "node". The lib should work in a browser environment as well. Is that possible? See also #47

**Review by alber70g — CHANGES_REQUESTED** [review]

File: packages/libs/chainwebjs/src/config.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import { IConfig } from './types';
+
+export const config: IConfig = {
+  network: 'mainnet01',
+  host: 'https://api.chainweb.com',
+};
```
[the remark this answers is not in the document]
Comment (KdWarlordz) [reply packages/libs/chainwebjs/src/config.ts:6]: i like this, will do

**Review by alber70g — APPROVED** [review]

**Review by ash-vd — APPROVED** [review]

**Pull request merged**